tests/ignite/metrics/GAN/test_fid.py

```diff
@@ -0,0 +1,175 @@
+from ignite.metrics.GAN.fid import FID, InceptionExtractor, fid_score
+
+
+def test_fid_function():
+    train_samples, test_samples = torch.rand(10, 10), torch.rand(10, 10)
+
+    mu1, sigma1 = train_samples.mean(axis=0), cov(train_samples, rowvar=False)
+    mu2, sigma2 = test_samples.mean(axis=0), cov(test_samples, rowvar=False)
+
+    assert pytest.approx(
+        fid_score(mu1, mu2, torch.tensor(sigma1, dtype=torch.float64), torch.tensor(sigma2, dtype=torch.float64))
+    ) == pytorch_fid_score.calculate_frechet_distance(mu1, sigma1, mu2, sigma2)
+
+
+def test_compute_fid_from_features():
+    train_samples, test_samples = torch.rand(10, 10), torch.rand(10, 10)
+
+    fid_scorer = FID(num_features=10)
+    fid_scorer.update([train_samples[:5], test_samples[:5]])
+    fid_scorer.update([train_samples[5:], test_samples[5:]])
+
+    mu1, sigma1 = train_samples.mean(axis=0), cov(train_samples, rowvar=False)
+    mu2, sigma2 = test_samples.mean(axis=0), cov(test_samples, rowvar=False)
+
+    assert pytest.approx(pytorch_fid_score.calculate_frechet_distance(mu1, sigma1, mu2, sigma2)) == fid_scorer.compute()
+
+
+def test_wrong_inputs():
+    with pytest.raises(ValueError, match=r"num of features must be greater to zero"):
+        FID(num_features=-1)
+    with pytest.raises(ValueError, match=r"Features must be a tensor of dim 2 \(got: 1\)"):
+        FID(num_features=1).update(torch.Tensor([[], []]))
+    with pytest.raises(ValueError, match=r"Batch size should be greater than one \(got: 0\)"):
+        FID(num_features=1).update(torch.rand(2, 0, 0))
+    with pytest.raises(ValueError, match=r"Feature size should be greater than one \(got: 0\)"):
+        FID(num_features=1).update(torch.rand(2, 2, 0))
+
+
+def test_statistics():
+    train_samples, test_samples = torch.rand(10, 10), torch.rand(10, 10)
+    fid_scorer = FID(num_features=10)
+    fid_scorer.update([train_samples[:5], test_samples[:5]])
+    fid_scorer.update([train_samples[5:], test_samples[5:]])
+
+    mu1, sigma1 = train_samples.mean(axis=0), torch.tensor(cov(train_samples, rowvar=False))
+    mu2, sigma2 = test_samples.mean(axis=0), torch.tensor(cov(test_samples, rowvar=False))
+
+    fid_mu1 = fid_scorer._train_mu
+    fid_sigma1 = fid_scorer._train_sigma / (fid_scorer._num_examples - 1)
+
+    fid_mu2 = fid_scorer._test_mu
+    fid_sigma2 = fid_scorer._test_sigma / (fid_scorer._num_examples - 1)
+
+    assert torch.isclose(mu1.double(), fid_mu1).all()
+    for cov1, cov2 in zip(sigma1, fid_sigma1):
+        assert torch.isclose(cov1.double(), cov2).all()
+    assert torch.isclose(mu2.double(), fid_mu2).all()
+    for cov1, cov2 in zip(sigma2, fid_sigma2):
+        assert torch.isclose(cov1.double(), cov2).all()
+
+
+def test_inception_extractor_wrong_inputs():
+    with pytest.raises(ValueError, match=r"Images should be of size 3x299x299 \(got torch.Size\(\[2, 2, 2, 0\]\)\)"):
+        InceptionExtractor()(torch.rand(2, 2, 2, 0))
+
+
+def _test_distrib_integration(device):
```

Comment:
This test is not distributed... have a look to others metric tests 

https://github.com/pytorch/ignite/blob/dce97dfeacf1516e78b1cd9c6a37e57a1228ceed/tests/ignite/metrics/test_precision.py#L402